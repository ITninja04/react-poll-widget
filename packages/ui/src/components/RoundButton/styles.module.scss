@import "../../colors";

.round_button {
  width: 4rem;
  height: 4rem;

  background: $darkblue;
  border-radius: 2rem;
  border: none;

  & #icon {
    width: 4rem;
    height: 4rem;
  }
}
