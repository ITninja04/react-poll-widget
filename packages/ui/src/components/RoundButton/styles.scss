@import "../../colors";
@import "../../mixin";

.ui.button.round-button {
  position: relative;
  display: inline-flex;
  align-items: center;
  transition: $transition-duration;

  &::after {
    @include shadow-1;
    position: absolute;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    border-radius: 200px;

    transition: $transition-duration;
    content: "";
  }

  &:hover {
    &::after {
      @include shadow-2;
    }

    .close-button {
      opacity: 1;
      background: transparent !important;
    }
  }

  h5 {
    font-size: 20px;
    margin-top: 0;
    transition: $transition-duration;

    &.collapsed {
      transform: translateX(50%) translateZ(0rem);
      opacity: 0;
    }
  }

  .text-box {
    display: flex;
    align-items: center;
    width: 0;
    margin: 0 0.5em;
    transition: $transition-duration;
    white-space: nowrap;

    &.collapsed {
      margin: 0;
    }
  }

  i.icon {
    display: inline-flex;
    align-items: center;
    justify-content: center;
    height: 100%;
    color: $white;
  }

  .close-button {
    position: absolute;
    top: -25%;
    right: -12.5%;
    opacity: 0;

    box-shadow: none;
    transition: $transition-duration;
  }
}
