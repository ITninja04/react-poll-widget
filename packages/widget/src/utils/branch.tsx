export type BranchFunction = (condition: boolean, ComponentIfTrue: React.ReactNode, ComponentIfFalse: React.ReactNode) => React.ReactNode;

export const branch: BranchFunction = (condition, ComponentIfTrue, ComponentIfFalse) => condition
  ? ComponentIfTrue
  : ComponentIfFalse;