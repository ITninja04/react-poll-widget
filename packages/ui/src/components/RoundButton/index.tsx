import React from 'react';
import { Button, Transition } from 'semantic-ui-react';
import { withState, withHandlers, compose } from 'recompose';

import './styles.scss';

const RoundButton: React.FC<{
  children: React.ReactNode,
  animate: boolean,
  toggleAnimation: () => void
}> = ({ children, animate, toggleAnimation }) => (
  <Button
    circular
    icon
    size='huge'
    color='blue'
    onClick={toggleAnimation}
    className='round-button'
  >
    <Transition animation='jiggle' visible={animate}>
      {children}
    </Transition>
  </Button>
)

export default compose(
  withState('animate', 'setAnimate', true),
  withHandlers({
    toggleAnimation: ({ animate, setAnimate }) => () => setAnimate(!animate)
  })
)(RoundButton)