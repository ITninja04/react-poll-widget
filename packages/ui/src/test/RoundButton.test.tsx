import React from "react";
import { Icon } from "semantic-ui-react";

import RoundButton from "../components/RoundButton";
import { createSnapshotTest } from "./utils";

createSnapshotTest(RoundButton, {
  color: 'red',
  Icon: <Icon name='chart bar' />
}, 'Basic')

createSnapshotTest(RoundButton, {
  color: 'red',
  Icon: <Icon name='chart bar' />,
  withCloseButton: true
}, 'With close button')

createSnapshotTest(RoundButton, {
  color: 'red',
  Icon: <Icon name='chart bar' />,
  text: 'Long test'
}, 'With content')