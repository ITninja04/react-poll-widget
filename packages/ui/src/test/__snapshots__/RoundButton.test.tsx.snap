// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Snapshot) Basic should render 1`] = `
<DocumentFragment>
  <button
    class="ui red huge circular icon button round-button"
  >
    <i
      aria-hidden="true"
      class="chart bar icon jiggle"
    />
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) With close button should render 1`] = `
<DocumentFragment>
  <button
    class="ui red huge circular icon button round-button"
  >
    <i
      aria-hidden="true"
      class="chart bar icon jiggle"
    />
  </button>
  <button
    class="ui basic icon button close-button"
  >
    <i
      aria-hidden="true"
      class="close icon"
    />
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) With content should render 1`] = `
<DocumentFragment>
  <button
    class="ui red huge circular icon button round-button"
  >
    <i
      aria-hidden="true"
      class="chart bar icon jiggle"
    />
    <div
      class="text-box collapsed"
      style="padding-right: 0px;"
    >
      <h5
        class="collapsed"
      >
        Long test
      </h5>
    </div>
  </button>
</DocumentFragment>
`;
